File: coinbase-feed/src/domain/entities/ticker-event.ts
import { TickerEventType } from "common/common/enums";

export class TickerEvent {
  time: Date;
  pair: string;
  price: number;
  type: TickerEventType;
  size: number;
  readonly exchange: string = "Coinbase Pro";

  constructor(data: {time: Date, pair: string, price: number, type: TickerEventType, size: number} ) {
    this.time = data.time;
    this.pair = data.pair;
    this.price = data.price;
    this.type = data.type;
    this.size = data.size;
  }
}

File: coinbase-feed-golang/src/common/types/ticker_event.go
package types

type TickerEvent struct {
	Time     string  `json:"time"`
	Pair     string  `json:"pair"`
	Price    float64 `json:"price"`
	Type     string  `json:"type"`
	Size     float64 `json:"size"`
	Exchange string  `json:"exchange"`
}
